### PR TITLE
Fix Flatbuffer build issue.

### PR DESCRIPTION
**Pull request by phoebesv**
### Description

Building on g++ 13.2.0 results in -Wstringop-overread errors on Linux. This commit addresses the flatbuffer build issue with the following changes:
1. Remove the Werror flag in the flarbuffer patch.
2. Add a compilation option to suppress the 'stringop-overflow' error in the Flatbuffers within the xnnpack provider.

### Motivation and Context
https://github.com/google/flatbuffers/issues/8119
https://github.com/microsoft/onnxruntime/pull/19239

### CODE REVIEW

**Comment by phoebesv** [comment]
@snnn @yihonglyu, PTAL, thanks!

File: cmake/onnxruntime_providers_xnnpack.cmake [hunk 1]
```diff
@@ -19,6 +19,9 @@
     flatbuffers::flatbuffers Boost::mp11 safeint_interface
   )
 
+  # Add compile option to suppress stringop-overflow error in Flatbuffers.
+  target_compile_options(onnxruntime_providers_xnnpack PRIVATE -Wno-error=stringop-overflow)
```
[the remark this answers is not in the document]
Comment (phoebesv) [reply cmake/onnxruntime_providers_xnnpack.cmake]: done, thx!

**Review by snnn — APPROVED** [review]

**Pull request merged**